js/src/tests/test/run.py

```diff
@@ -1,24 +1,76 @@
 #!/usr/bin/env python
 # Adapted from https://github.com/tc39/test262/blob/master/tools/generation/test/run.py
 
-import shutil, subprocess, sys, os, unittest
+import shutil, subprocess, contextlib, tempfile, sys, os, unittest
 
 testDir = os.path.dirname(os.path.relpath(__file__))
 OUT_DIR = os.path.join(testDir, 'out')
 EXPECTED_DIR = os.path.join(testDir, 'expected')
 ex = os.path.join(testDir, '..', 'test262-export.py')
+importExec = os.path.join(testDir, '..', 'test262-update.py')
+test262Url = 'git://github.com/tc39/test262.git'
+
+@contextlib.contextmanager
+def TemporaryDirectory():
+    tmpDir = tempfile.mkdtemp()
+    try:
+        yield tmpDir
+    finally:
+        shutil.rmtree(tmpDir)
 
 class TestExport(unittest.TestCase):
     maxDiff = None
 
-    def fixture(self, name):
-        relpath = os.path.relpath(os.path.join(testDir, 'fixtures', name))
+    def exportScript(self):
+        relpath = os.path.relpath(os.path.join(testDir, 'fixtures', 'export'))
         sp = subprocess.Popen(
             [ex, relpath, '--out', OUT_DIR],
             stdout=subprocess.PIPE)
         stdout, stderr = sp.communicate()
         return dict(stdout=stdout, stderr=stderr, returncode=sp.returncode)
 
+    def importLocal(self):
+        with TemporaryDirectory() as cloneDir:
+            branch = 'smTempBranch'
+            # Clone Test262 to a local branch
+            subprocess.check_call(
+                ['git', 'clone', '--depth=1', test262Url, cloneDir]
+            )
+            # Checkout to a new branch
+            subprocess.check_call(
+                ['git', '-C', cloneDir, 'checkout', '-b', branch]
+            )
+            # Make changes on the new branch
+            # Remove test/language/export/escaped-from.js
+            subprocess.check_call(
+                ['git', '-C', cloneDir, 'rm',
+                    'test/language/export/escaped-from.js']
+            )
+            # Rename test/language/export/escaped-default.js
+            subprocess.check_call(
+                ['git', '-C', cloneDir, 'mv',
+                    'test/language/export/escaped-default.js',
+                    'test/language/export/escaped-foobarbaz.js',
+                ]
+            )
+            # Copy fixtures files
+            fixturesDir = os.path.join(testDir, 'fixtures', 'import', 'files')
+            shutil.copytree(fixturesDir, os.path.join(cloneDir, 'test', 'temp42'))
+            # Stage and Commit changes
+            subprocess.check_call((['git', '-C', cloneDir, 'add', '.']))
+            subprocess.check_call((['git', '-C', cloneDir, 'commit', '-m', '"local foo"']))
```

Comment:
_you listen the sound resonating in the room. It loudly spells a git sorcery, not louder but neither fading as a noisy "local foo"_